Finmer.Editor/FormDocumentScene.cs

```diff
@@ -510,6 +510,49 @@ private void trvNodes_DragDrop(object sender, DragEventArgs e)
             Dirty = true;
         }
 
+        public TreeNode GetTreeNodeByKey(string key) 
+        {
+            //Prepare stack with root TreeNodes in it
+            Stack<TreeNode> stack = new Stack<TreeNode>();
+            foreach (TreeNode root_node in trvNodes.Nodes)
+                stack.Push(root_node);
+
+            // Flood-fill down the tree until the requested node is found
+            while (stack.Count > 0)
+            {
+                TreeNode node = stack.Pop();
+                var node_sn = (AssetScene.SceneNode)node.Tag;
+
+                // Links do not have keys
+                if (node_sn.IsLink)
+                    continue;
+
+                // Is this the node we're looking for?
+                if (node_sn.Key.Equals(key))
+                    return node;
+
+                // Recursively examine the node's children
+                foreach (TreeNode child in node.Nodes)
+                    stack.Push(child);
+            }
+
+            return null;
+        }
+
+        private void trvNodes_Link_DoubleClick(object sender, TreeNodeMouseClickEventArgs e) 
+        {
+            //Find the type of node that is double-clicked on, and see if it's a link node
+            var link_node = e.Node;
+            var link_node_sn = (AssetScene.SceneNode)link_node.Tag;
+            if (link_node_sn.IsLink)
+            {
+                //If node is a link node, jump to the node it's linked to.
+                var search_key = link_node_sn.LinkTarget;
+                var matching_node = GetTreeNodeByKey(search_key);
+                trvNodes.SelectedNode = matching_node;
```

Comment:
`GetTreeNodeByKey` may return `null`. In that case, we'll assign `null` to `SelectedNode`, causing the link to be deselected. Is that the behavior you intend?